internal/wire/testdata/Generic/foo/wire.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 The Wire Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build wireinject
+// +build wireinject
+
+// wire.go
+package main
+
+import "github.com/google/wire"
+
+func InitializeFooStore() FooStorer {
```

Comment:
This is a good start. We should add to this test to ensure Wire can also generate injector functions with generic-typed parameters and return values correctly. If all that works, then LGTM.